torchvision/prototype/datasets/_builtin/coco.py

```diff
@@ -26,24 +28,144 @@
 from torchvision.prototype.datasets.utils._internal import (
     MappingIterator,
     INFINITE_BUFFER_SIZE,
+    BUILTIN_DIR,
     getitem,
     path_accessor,
     path_comparator,
 )
+from torchvision.prototype.features import BoundingBox, Label
+from torchvision.prototype.features._feature import DEFAULT
+from torchvision.prototype.utils._internal import FrozenMapping
 
-HERE = pathlib.Path(__file__).parent
+
+class CocoLabel(Label):
+    super_category: Optional[str]
+
+    @classmethod
+    def _parse_meta_data(
+        cls,
+        category: Optional[str] = DEFAULT,  # type: ignore[assignment]
+        super_category: Optional[str] = DEFAULT,  # type: ignore[assignment]
+    ) -> Dict[str, Tuple[Any, Any]]:
+        return dict(category=(category, None), super_category=(super_category, None))
+
+
+# TODO: this is in desperate need of comments. They will be added as soon as we are sure this is the way to go
+class CocoAnnotationsCollater(IterDataPipe[Tuple[Dict[str, Any], Dict[str, Dict[str, Any]]]]):
```

Comment:
Is this re-implementing what `pycocotools` do internally, to avoid a dependency?

Reply:
Kind of, but not to avoid a dependency. I'm not sure if it works well with datapipes. Will have a closer look. Unless we want to avoid the dependency?

Reply:
We maybe can use it, but it depends how far we want to go into bad practices. The `COCO` object loads the file with `open` which fails for file handles coming from a zip file. Patching this would be fairly easy using something like `unittest.mock`, but that is kind of fragile, because we can't treat `pycocotools` as black box anymore. That being said, I'm fairly certain that newer versions of `pycocotools` won't break our patch, because it is already really stable and this is something which is most likely not going to be touched in newer versions.

Reply:
There are basically 3 things that `pycocotools` do for us which are helpful:
- load the data (what you are doing here)
- convert polygons to segmentation maps (with C++ extensions for that)
- evaluation code

I would like the evaluation code to also be part of torchvision sometime in the future, as it would enable us to avoid having to hold the COCO json in memory (and we would be able to perform progressive evaluation in a more streamlined fashion, in a better way than what we do in the `references/` for now).

This would leave us with just the polygons conversion, which is a very handy tool but we could implement it in torchvision as well.

So ultimately I think that we might want to drop a dependency on `pycocotools`, but it will require further work